lib/f_core/net/network/c_ipv4.cpp

```diff
@@ -0,0 +1,50 @@
+// Self Include
+#include "f_core/net/network/c_ipv4.h"
+
+// Zephyr Includes
+#include <zephyr/logging/log.h>
+#include <zephyr/net/net_if.h>
+
+LOG_MODULE_REGISTER(CIPv4);
+
+CIPv4::CIPv4(const char* ip) : ip(ip), netIface(*net_if_get_default()) {
```

Comment:
I think you could delegate this constructor to `CIPv4::CIPv4(const char* ip, net_if *net_iface)`
```C++
CIPv4::CIPv4(const char* ip) : CIPv4(ip, *net_if_get_default()) {}
```
so you dont have duplicate construction logic. (here its a single function call so its no biggie but could be alil cleaner)

Reply:
Fair